### PR TITLE
#33 - Added PathPrefixSlice

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #33 
Added `PathPrefixSlice`

### CODE REVIEW

File: src/main/java/com/artipie/http/client/PathPrefixSlice.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        final Iterable<Map.Entry<String, String>> headers,
+        final Publisher<ByteBuffer> body
+    ) {
+        final RequestLineFrom rqline = new RequestLineFrom(line);
+        final URI uri = rqline.uri();
+        return this.origin.response(
+            new RequestLine(
+                rqline.method().value(),
+                new URIBuilder(uri)
+                    .setPath(String.format("%s%s", this.prefix, uri.getPath()))
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/http/client/PathPrefixSlice.java:76]: @olegmoz hm, I do not quite understand why create `URIBuilder` from original `uri` and then call `setPath`... Why cannot we simply call `URI.create(String.format("%s%s", this.prefix, uri.getPath()))` and inline `uri` variable?

File: src/test/java/com/artipie/http/client/PathPrefixSliceTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                MatcherAssert.assertThat(
+                    "Body is preserved",
+                    new PublisherAs(rqbody).bytes().toCompletableFuture().join(),
+                    new IsEqual<>(body)
+                );
+                return StandardRs.OK;
+            },
+            "/prefix"
+        ).response(
+            new RequestLine(method, "/path").toString(),
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/http/client/PathPrefixSliceTest.java:76]: @olegmoz I'd suggest to parametrize this test with different paths and prefixes: at least add complex path and prefix and `/` path

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz added two suggestions, please, take a look

File: src/main/java/com/artipie/http/client/PathPrefixSlice.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/http/client/PathPrefixSlice.java:76]: @olegmoz generally URI may contain host and port for proxy requests, but we do not really support it anyway. So I will simplify it, thanks

Comment (olegmoz) [reply src/main/java/com/artipie/http/client/PathPrefixSlice.java:76]: @olenagerasimova jsut understood that there is other reason actually! we need to preserve query as well, so I guess `new URIBuilder(uri)` should be preserved

**Comment by olegmoz** [comment]
@olenagerasimova thanks to your comments I understood that there is no support for query in this class which is needed. So I supported query and improved tests, please review

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, looks good to merge

**Pull request merged**